shub/settings/config.py

```diff
@@ -159,3 +179,89 @@
     #    'globus',
     #    'saml_auth'
 ]
+
+# Any plugins enabled from the environment?
+PLUGINS_ENABLED += get_sregistry_envar_list("SREGISTRY_PLUGINS_ENABLED")
+
+## Environment
+
+# Custom additions are handled before this
+# Limit defaults that user can set
+BOOLEAN_DEFAULTS = [
+    "ENABLE_GOOGLE_AUTH",
+    "ENABLE_TWITTER_AUTH",
+    "ENABLE_GITHUB_AUTH",
+    "ENABLE_GITLAB_AUTH",
+    "ENABLE_BITBUCKET_AUTH",
+    "ENABLE_GITHUB_ENTERPRISE_AUTH",
+    "USER_COLLECTIONS",
+    "PRIVATE_ONLY",
+    "DEFAULT_PRIVATE",
+    "DISABLE_BUILDING",
+    "SREGISTRY_GOOGLE_BUILD_SINGULARITY_VERSION",
+    "DISABLE_GITHUB",
+    "DISABLE_BUILD_RECEIVE",
+    "MINIO_SSL",
+    "MINIO_MULTIPART_UPLOAD",
+    "DISABLE_MINIO_CLEANUP",
+    "LOGGING_SAVE_RESPONSES",
+    "VIEW_RATE_LIMIT_BLOCK",
+]
+
+STRING_DEFAULTS = [
+    "DOMAIN_NAME",
+    "DOMAIN_NAME_HTTP",
+    "HELP_CONTACT_EMAIL",
+    "HELP_INSTITUTION_SITE",
+    "REGISTRY_NAME",
+    "REGISTRY_URI",
+    "GOOGLE_ANALYTICS",
+    "MINIO_SERVER",
+    "MINIO_EXTERNAL_SERVER",
+    "MINIO_BUCKET",
+    "MINIO_REGION",
+    "VIEW_RATE_LIMIT",
+]
+
+INTEGER_DEFAULTS = [
+    "DATA_UPLOAD_MAX_MEMORY_SIZE",
+    "USER_COLLECTION_LIMIT",
+    "COLLECTIONS_VIEW_PAGE_COUNT",
+    "CONTAINER_WEEKLY_GET_LIMIT",
+    "COLLECTION_WEEKLY_GET_LIMIT",
+    "SREGISTRY_GOOGLE_BUILD_LIMIT",
+    "SREGISTRY_GOOGLE_BUILD_TIMEOUT_SECONDS",
+    "SREGISTRY_GOOGLE_BUILD_EXPIRE_SECONDS",
+    "CONTAINER_SIGNED_URL_EXPIRE_SECONDS",
+    "MINIO_SIGNED_URL_EXPIRE_MINUTES",
+]
+
+# Set envars by type
+for key in BOOLEAN_DEFAULTS:
+    value = get_sregistry_envar(key)
+
+    # Setting a boolean
+    if value in ["true", "True"]:
+        locals()[key] = True
+        print("Setting %s from the environment to True" % key)
+
+    elif value in ["False", "false"]:
+        locals()[key] = False
+        print("Setting %s from the environment to False" % key)
+
+# Integer values
+for key in INTEGER_DEFAULTS:
+    value = get_sregistry_envar(key)
+    if value:
+        try:
+            locals()[key] = int(value)
+        except:
+            print("There was an issue setting %s as an integer." % key)
```

Comment:
It would be better to either let this fail or re-raise an exception so that the container fails rather than continue with an unexpectedly invalid value.